### DISCUSSION TITLE
how to remove console.log in production in nuxt3

### DISCUSSION THREAD

**Discussion by Max199909** [nuxt/nuxt#22625] [Questions]
how to remove console.log in production in nuxt3

**Comment by logotip4ik** [comment]
Nuxt uses Vite under the hood, and Vite exposes esbuild options. You could use these options to hint esbuild that `console.log` is safe to drop.

Little code example ([source](https://github.com/vitejs/vite/discussions/7920#discussioncomment-3996695)):

```js
// nuxt.config.ts
export default defineNuxtConfig({
  vite: {
    esbuild: {
      drop: ['debugger'],
      pure: ['console.log', 'console.error', 'console.warn', 'console.debug', 'console.trace'],
    },
  },
})
```

**Reply by xjccc** [reply]
[vite-plugin-remove-console](https://www.npmjs.com/package/vite-plugin-remove-console) is awesome

**Comment by forgetso** [comment]
> Nuxt uses Vite under the hood, and Vite exposes esbuild options. You could use these options to hint esbuild that `console.log` is safe to drop.
> 
> Little code example ([source](https://github.com/vitejs/vite/discussions/7920#discussioncomment-3996695)):
> 
> ```js
> // nuxt.config.ts
> export default defineNuxtConfig({
>   vite: {
>     esbuild: {
>       drop: ['debugger'],
>       pure: ['console.log', 'console.error', 'console.warn', 'console.debug', 'console.trace'],
>     },
>   },
> })
> ```

This pure stuff doesn't work for me. I still get warns from external deps.
